app/shared/src/commonMain/kotlin/ui/subject/episode/details/components/PaginatedEpisodeList.kt

```diff
@@ -0,0 +1,261 @@
+                            color = MaterialTheme.colorScheme.onSurface,
+                            modifier = Modifier.padding(end = 4.dp),
+                            maxLines = 1
+                        )
+                        Icon(
+                            Icons.Outlined.ArrowDropDown,
+                            contentDescription = "选择分组",
+                            tint = MaterialTheme.colorScheme.onSurfaceVariant,
+                        )
+                    }
+                }
+
+                DropdownMenu(
+                    expanded = showGroupSelector,
+                    onDismissRequest = { showGroupSelector = false },
+                ) {
+                    episodeGroups.forEachIndexed { index, group ->
+                        DropdownMenuItem(
+                            text = {
+                                Text(
+                                    group.title,
+                                    color = if (index == currentGroupIndex) MaterialTheme.colorScheme.primary else MaterialTheme.colorScheme.onSurface,
+                                )
+                            },
+                            onClick = {
+                                currentGroupIndex = index
+                                showGroupSelector = false
+                            },
+                        )
+                    }
+                }
+            }
+
+            IconButton(
+                onClick = {
+                    if (currentGroupIndex < episodeGroups.size - 1) {
+                        currentGroupIndex++
+                    }
+                },
+                enabled = currentGroupIndex < episodeGroups.size - 1,
+            ) {
+                Icon(
+                    Icons.Outlined.ChevronRight,
+                    contentDescription = "下一组",
+                    tint = if (currentGroupIndex < episodeGroups.size - 1) MaterialTheme.colorScheme.onSurface else MaterialTheme.colorScheme.onSurface.copy(
+                        alpha = 0.38f,
+                    ),
+                )
+            }
+        }
+
+        // Episode List with all episodes and group headers
+        LazyColumn(
+            state = listState,
+            verticalArrangement = Arrangement.spacedBy(4.dp),
+            modifier = Modifier.weight(1f),
+            contentPadding = PaddingValues(horizontal = 16.dp, vertical = 8.dp),
+        ) {
+            episodeGroups.forEachIndexed { groupIndex, group ->
+                item(key = "header_$groupIndex") {
+                    Text(
+                        group.title,
+                        style = MaterialTheme.typography.titleSmall,
+                        color = MaterialTheme.colorScheme.primary,
+                        modifier = Modifier.padding(vertical = 8.dp)
+                    )
+                }
+
+                items(
+                    items = group.episodes,
+                    key = { "${groupIndex}_${it.episodeId}" }
```

Comment:
所有的 key 必须 prefix 一个全局唯一的且方便搜索的名字
```suggestion
                    key = { "PaginatedEpisodeList_${groupIndex}_${it.episodeId}" }
```